src/OpenTelemetry.Instrumentation.Dependencies/TracerBuilderExtensions.cs

```diff
@@ -71,8 +71,27 @@ public static TracerBuilder AddDependencyInstrumentation(this TracerBuilder buil
                 .AddInstrumentation((t) => new AzureClientsInstrumentation(t))
                 .AddInstrumentation((t) => new AzurePipelineInstrumentation(t))
                 .AddInstrumentation((t) => new HttpClientInstrumentation(t, httpOptions))
-                .AddInstrumentation((t) => new HttpWebRequestInstrumentation(t, httpOptions))
                 .AddInstrumentation((t) => new SqlClientInstrumentation(t, sqlOptions));
         }
+
+        /// <summary>
+        /// Enables the outgoing requests automatic data collection.
```

Comment:
lets make the comment more explicitly state that this enables outgoing calls made by HttpWebRequest, and is applicable in .NET Framework only.. something of that tone.
Maybe rename the method itself to be more specific about httpwebrequest.

We can later add a common AddAllDependencyInstrumentation helper covering all dependencies - thoughts?

Reply:
My thinking here was this one call would spin up all our dependency instrumentation, similar to what `public static TracerBuilder AddDependencyInstrumentation(this TracerBuilder builder)` does today. Right now it just has the one .NET Framework thing because we haven't converted the others, but that will be changing right?

It should look something like this at the end of the day:

```csharp
        public static OpenTelemetryBuilder AddDependencyInstrumentation(
            this OpenTelemetryBuilder builder,
            Action<HttpClientInstrumentationOptions> configureHttpInstrumentationOptions = null,
            Action<SqlClientInstrumentationOptions> configureSqlInstrumentationOptions = null)
        {
            if (builder == null)
            {
                throw new ArgumentNullException(nameof(builder));
            }

            builder.AddActivitySource(AzureClientsActivitySource.ActivitySourceName);
            builder.AddActivitySource(AzurePipelineActivitySource.ActivitySourceName);
            builder.AddActivitySource<HttpClientInstrumentationOptions>(HttpClientActivitySource.ActivitySourceName, configureHttpInstrumentationOptions);
            builder.AddActivitySource<SqlClientInstrumentationOptions>(SqlClientActivitySource.ActivitySourceName, configureSqlInstrumentationOptions);
#if NET461
            GC.KeepAlive(HttpWebRequestActivitySource.Instance);

            builder.AddActivitySource(HttpWebRequestActivitySource.ActivitySourceName);
#endif
            return builder;
        }
```

So it's not specific to .NET Framework. I also invented a way to pass options, looks like we have a gap there.

If we had this, then we would remove the existing `TraceBuilder` extensions?

Reply:
I actually to say we need to have separate methods for each dependency - like AddSqlDependencyInstrumentation, AddHttpClientInstrumentation etc. So users can pick individual ones if they want.

We'll also have a method like AddDependencyInstrumentation() which most users would use. (exactly the one you shared above).

(having separate method allows us to turn on/off individual collections. very helpful when investigating some leak or issues, and want to isolate to a particular instrumentation)

Reply:
And you are right about removing TracerBuilder extension method- Once we finish span->activity work, there would be no TracerBuilder!

Reply:
I split the new stuff out into OpenTelemetryBuilderExtensions.cs and added a method for registering all sources (which will grow) or the HttpWebRequest source only. LMK.